cmd/create/admin/cmd.go

```diff
@@ -222,3 +205,53 @@ func generateRandomPassword(length int) (string, error) {
 
 	return string(pw), nil
 }
+
+func FindExistingClusterAdminIDP(cluster *cmv1.Cluster, r *rosa.Runtime) (
+	htpasswdIDP *cmv1.IdentityProvider, userList *cmv1.HTPasswdUserList) {
+
+	// admin user will now be created in a htpasswd IDP named 'cluster-admin'
+	// It should suffice to look for this idp but for back-compatibility ,
+	// i.e for cases where the cluster was created  with previous versions of rosacli
+	// we are going to search  all htpasswd idps
+
+	r.Reporter.Debugf("Loading cluster's identity providers")
+	r.Reporter.Infof("Loading cluster's identity providers")
+	idps, err := r.OCMClient.GetIdentityProviders(cluster.ID())
+	if err != nil {
+		r.Reporter.Errorf("Failed to get identity providers for cluster '%s': %v", r.ClusterKey, err)
+		os.Exit(1)
+	}
+
+	for _, item := range idps {
+		if ocm.IdentityProviderType(item) == ocm.HTPasswdIDPType {
+
+			itemUserList, err := r.OCMClient.GetHTPasswdUserList(cluster.ID(), item.ID())
+			r.Reporter.Infof("user list %s: %v", item.Name(), itemUserList)
+			if err != nil {
+				r.Reporter.Errorf("Failed to get user list of the HTPasswd IDP of '%s: %s': %v", item.Name(), r.ClusterKey, err)
+				os.Exit(1)
+			}
+			if HasClusterAdmin(itemUserList) {
+				r.Reporter.Infof("Has admintrue")
+
+				htpasswdIDP = item
+				userList = itemUserList
+				return
+			} else {
+				r.Reporter.Infof("Has adminfalse")
+			}
+		}
+	}
+	return
+}
+
+func HasClusterAdmin(userList *cmv1.HTPasswdUserList) bool {
+	hasAdmin := false
+	userList.Each(func(user *cmv1.HTPasswdUser) bool {
+		if user.Username() == ClusterAdminUsername {
+			hasAdmin = true
+		}
+		return true
+	})
+	return hasAdmin
+}
```

Comment:
I dont think we need to itterate the full list if we find a user with the admin user name, we should just return, can either update the current function : 
```
if user.Username() == ClusterAdminUsername {
			hasAdmin = true
			return false
		}
```
`Each` function will keep itterating till it exhaustes the list or until the func returns false. 

Alternatively we can just use `range` and return true from that :  
```suggestion
func HasClusterAdmin(userList *cmv1.HTPasswdUserList) bool {
	for _, user := range userList.Slice() {
		if user.Username() == ClusterAdminUsername {
			return true
		}
	}
	return false
}
```

What ever you think yourself

Reply:
This too is a remnant of the old code , just happened to  move it from one file to another.

I will fix that.